src/EventStore.Common/Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -16,15 +17,25 @@ public static class ConfigurationBuilderExtensions {
 
 			var configurationPath = root.GetValue<string>("Config");
 
+			var configurationPathOptional = CompareConfigFilePath(defaultValues, configurationPath);
 			var yamlSource = new YamlSource {
 				Path = configurationPath,
-				Optional = true,
+				Optional = configurationPathOptional,
 				ReloadOnChange = true
 			};
 			yamlSource.ResolveFileProvider();
 			builder.Sources.Insert(1, yamlSource);
 
 			return builder;
 		}
+
+		private static bool CompareConfigFilePath(IEnumerable<KeyValuePair<string, object?>> defaultValues, string? configurationPath) {
+			var defaultValuesDictionary = defaultValues.ToDictionary(x => x.Key, x => x.Value);
+
+			if (defaultValuesDictionary.TryGetValue("Config", out var defaultConfigurationPath)) {
+				return string.Equals(configurationPath, defaultConfigurationPath?.ToString());
```

Comment:
i think the ReferenceEquals that we had before would be preferable (perhaps with a comment), so that if the user happens to specify a config path that is the same as the default then they still get the advantage of this existence check

Reply:
actually, come to think of it, we have `CheckProvidersForEnvironmentVariables` in ConfigurationRootExtensions. Maybe we could add a method in there that checks if a value has been overridden from the default. that would be less cryptic than ReferenceEquals

Reply:
> i think the ReferenceEquals that we had before would be preferable (perhaps with a comment), so that if the user happens to specify a config path that is the same as the default then they still get the advantage of this existence check

The  default configuration ReferenceEquals check was failing while running the tests in Ubuntu. The following check supposed to return true but it's returning false and as per Logic I was seeing "Config not found" in the logs. 
ReferenceEquals(configurationPath, defaultConfigurationPath)